README: Use docker containers for tests

dcarley opened the pull request:
## What

I think this is preferable to using Homebrew because it:

- doesn't pollute my main machine with necessary dependencies
- is consistent with our documentation for Postgres

I had to tweak the instructions for getting the IP address of the Docker
host. For now we can assume that everyone is using Docker Machine, but they
will need to use the appropriate name of their VM.

Also the containers can be stopped and deleted in one command.

## How to review

Follow the instructions in the README and check whether it works for you.

## Who can review

Anyone

dcarley committed:
README: Don't list all env vars for tests

We didn't add the `MYSQL_*` environment variables to here and we'll probably
forget to update this if we add any more. So instead refer people to look at
the test suites directly.

dcarley committed:
README: Use container for local MySQL tests

I think this is preferable to using Homebrew because it:

- doesn't pollute my main machine with necessary dependencies
- is consistent with our documentation for Postgres

Unfortunately we have to wait for MySQL to have finished starting before we
can use it.

I've adjusted the instructions to suit Docker for Mac and Docker for Linux,
which don't need you to find the IP of the Docker VM or manually configure
port forwarding. We don't need to specify the hostname environment variables
because the default of `localhost` should work for everyone.

Finally, the containers can be stopped and deleted in one command.

paroxp commented on README.md:
```diff
@@ -115,30 +115,28 @@ There are two forms of tests for the broker, the unit tests and the integration
 
-brew install mysql
-mysql.server start
+docker run -p 3306:3306 --name mysql -e MYSQL_ALLOW_EMPTY_PASSWORD=yes -d mysql
```
Did you consider pinning the version we'd like to use for consistency with our environments?

Same applies to `postgres` I suppose.

dcarley replied:
Done now 👍

paroxp reviewed (commented):
I'm happy with the change as is, however being a little conscious with the version of services we're using.

paroxp commented on README.md:
```diff
@@ -115,30 +115,28 @@ There are two forms of tests for the broker, the unit tests and the integration
-brew install mysql
-mysql.server start
+docker run -p 3306:3306 --name mysql -e MYSQL_ALLOW_EMPTY_PASSWORD=yes -d mysql
+export MYSQL_HOSTNAME=${DOCKER_IP}
```
This turns out to be a `localhost` instead of the `DOCKER_IP` on my machine.

Is there more magic you've done in the mean time?

Again, similarly for `postgres` IMO.

alext replied:
I think this will vary depending on whether you're using docker-machine or Docker for Mac. The latter works a lot more like Docker on linux, and hides the fact that there's an underlying VM much more.

LeePorte replied:
I'd suggest using the `mysql/mysql-server:5.7` container so it matches our expected version.

dcarley replied:
@LeePorte Do you know if there's any difference between https://hub.docker.com/_/mysql/ and https://hub.docker.com/r/mysql/mysql-server/ ?

dcarley replied:
I've adjusted the instructions to work on Docker for Mac and Docker for Linux where you don't need to manually setup port forwarding.

dcarley committed:
README: Pin container versions for local tests

I have taken the versions from the catalog used by the CI tests:

- https://github.com/alphagov/paas-rds-broker/blob/5097bb8201f89428fbe7ae42f32121242183bab9/ci/blackbox/config.json
